src/bandersnatch/tests/test_main.py

```diff
@@ -54,23 +59,23 @@ def test_main_cant_create_config(caplog, tmpdir):
     assert not conf_path.exists()
 
 
-def test_main_reads_config_values(mirror_mock: mock.MagicMock, tmpdir):
+def test_main_reads_config_values(mirror_mock: mock.MagicMock, tmpdir: Path) -> None:
     base_config_path = Path(bandersnatch.__file__).parent / "unittest.conf"
-    diff_file = Path(tempfile.gettempdir()).joinpath("srv/pypi/mirrored-files")
+    diff_file = Path(tempfile.gettempdir()) / "srv/pypi/mirrored-files"
     config_lines = [
         f"diff-file = {diff_file.as_posix()}\n"
         if line.startswith("diff-file")
         else line
         for line in base_config_path.read_text().splitlines()
     ]
-    config_path = tmpdir.join("unittest.conf")
+    config_path = tmpdir / "unittest.conf"
     config_path.write_text("\n".join(config_lines), encoding="utf-8")
     sys.argv = ["bandersnatch", "-c", str(config_path), "mirror"]
     assert config_path.exists()
     main(asyncio.new_event_loop())
     (homedir, master), kwargs = mirror_mock.call_args_list[0]
 
-    assert "/srv/pypi" == homedir
+    assert Path("/srv/pypi") == homedir
```

Comment:
Thanks for this change on Mirror - just hope non fikeaystem objects can handle it - cc: @techalchemy